### PR TITLE
feature/FOUR-14548: Launchpad Setup - Embed Media - part 2

### PR DESCRIPTION
**Pull request by luNunezProcessmaker**
## Issue & Reproduction Steps
image-20240311-150141

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-14548

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy
ci:next

### CODE REVIEW

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 1]
```diff
@@ -1711,8 +1717,23 @@ public function saveImagesIntoMedia(Request $request, Process $process)
...
-                        $process->addMediaFromBase64($image['url'])->toMediaCollection('images_carousel');
+                    ->where('uuid', $image['uuid'])->exists()) {
+                        if ($image['type'] === self::CAROUSEL_FILE_TYPES['IMAGE']) {
+                            $process
+                            ->addMediaFromBase64($image['url'])
+                            ->withCustomProperties(['type' => $image['type']])
+                            ->toMediaCollection('images_carousel');
+                        }
+                        if ($image['type'] === self::CAROUSEL_FILE_TYPES['EMBED']) {
+                            $fakeFile = File::image('photo.jpg');
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php:1728]: Please upload a image in the `resources/img/launchpad-images/embed.png `
Use this image in this section download the image icon related to embed

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 2]
```diff
@@ -5,6 +5,7 @@
 use Illuminate\Contracts\Routing\ResponseFactory;
 use Illuminate\Http\Request;
 use Illuminate\Http\Response;
+use Illuminate\Http\Testing\File;
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php:8]: Remove this use

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 3]
```diff
@@ -41,6 +42,11 @@
 
 class ProcessController extends Controller
 {
+    const CAROUSEL_FILE_TYPES = [
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php:45]: Rename to CAROUSEL_TYPES

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: resources/js/components/shared/InputImageCarousel.vue [hunk 4]
```diff
@@ -403,16 +427,36 @@ export default {
...
+        return;
+      }
       this.images[index].url = this.embedUrls[index];
       this.unfocusIcon(index);
+      this.notURL = false;
+    },
+    isValidURL(urlString) {
+      let url;
+      try {
+        url = new URL(urlString);
```
Comment (fagubla) [inline resources/js/components/shared/InputImageCarousel.vue:450]: `const url = new URL(urlString)`

File: resources/js/components/shared/InputImageCarousel.vue [hunk 5]
```diff
@@ -403,16 +427,36 @@ export default {
...
+      } catch (error) {
+        return false;
+      }
+      return url.protocol === "http:" || url.protocol === "https:";
```
Comment (fagubla) [inline resources/js/components/shared/InputImageCarousel.vue:454]: move to section try

**Review by fagubla — CHANGES_REQUESTED** [review]

**Review by fagubla — APPROVED** [review]

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 6]
```diff
@@ -1711,8 +1718,23 @@ public function saveImagesIntoMedia(Request $request, Process $process)
             foreach ($request->imagesCarousel as $image) {
                 if (is_string($image['url']) && !empty($image['url'])) {
                     if (!$process->media()->where('collection_name', 'images_carousel')
-                        ->where('uuid', $image['uuid'])->exists()) {
-                        $process->addMediaFromBase64($image['url'])->toMediaCollection('images_carousel');
+                    ->where('uuid', $image['uuid'])->exists()) {
+                        if ($image['type'] === self::CAROUSEL_TYPES['IMAGE']) {
+                            $process
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php:1723]: Create a method to save the image in the media table

File: ProcessMaker/Http/Controllers/Api/ProcessController.php [hunk 7]
```diff
@@ -1711,8 +1718,23 @@ public function saveImagesIntoMedia(Request $request, Process $process)
...
+                            ->addMediaFromBase64($image['url'])
+                            ->withCustomProperties(['type' => $image['type']])
+                            ->toMediaCollection('images_carousel');
+                        }
+                        if ($image['type'] === self::CAROUSEL_TYPES['EMBED']) {
+                            $fakeFile =  File::image('photo.jpg');
+                            $process
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessController.php:1730]: Create a method to save the embed in the embed table

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: resources/js/components/shared/InputImageCarousel.vue [hunk 8]
```diff
@@ -263,41 +269,60 @@ export default {
...
+      this.embedUrls[this.images.length - 1] = mediaURL;
     },
     setProcessId(processId) {
       this.processId = processId;
     },
     getImages() {
       return this.images;
     },
+    /**
+     * Check if embed files are valid
```
Comment (fagubla) [inline resources/js/components/shared/InputImageCarousel.vue:314]: embed files?

**Review by fagubla — CHANGES_REQUESTED** [review]

**Review by fagubla — APPROVED** [review]

**Pull request closed without merging**